### PR TITLE
Move ResourceStatus to dynamic-plugin-sdk

### PR DESCRIPTION
**Pull request by fhlavac**
Moved ResourceStatus component to dynamic-plugin-sdk as a part of ListPage component migration effort.
https://issues.redhat.com/browse/HAC-311

@florkbr 
@andrewballantyne

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ import { Helmet } from 'react-helmet';
 import { Trans, useTranslation } from 'react-i18next';
 import { Link, match as RouterMatch } from 'react-router-dom';
 import { WatchK8sResource } from '@console/dynamic-plugin-sdk';
+import { ResourceStatus } from '@console/dynamic-plugin-sdk/src/app/components/utils/resource-status';
```
Comment (andrewballantyne) [inline frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx:19]: Any reason these are not root package exports? You gave one right above this line.

Comment (florkbr) [reply frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx:19]: Agree with @andrewballantyne here - we should not require consumers of the SDK to dig into the directory structure for components.

Would be nice to see:
```js
import { ResourceStatus, WatchK8sResource } from '@console/dynamic-plugin-sdk';
```

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
-import * as classNames from 'classnames';
 import { Badge } from '@patternfly/react-core';
+import * as classNames from 'classnames';
```
Comment (florkbr) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx:3]: @fhlavac was this change intentional?

Comment (fhlavac) [reply frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx:3]: No, it wasn't . Missed it in the diff. Thank you :slightly_smiling_face:

File: frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx (hunk 1 above)
Comment (fhlavac) [reply frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx]: @andrewballantyne Added them, thank you

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx (hunk 2 above)
Comment (andrewballantyne) [reply frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx:3]: Yeah, the public folder is not properly linted... intentionally and by design (old code, lots of changes would be needed). So moving it into `packages` folder does prompt auto-fixing lint errors, especially those around import statements being ordered.

**Review by andrewballantyne — APPROVED** [review]
Not much to it, moves the component over and updates references. This is a pretty low level component, so the change is pretty transparent (mostly) for the Console code.

No docs needed, moving/reorganizing code
/label docs-approved
/label px-approved

I tested a few pages with statuses to makes sure they still render fine
/label qe-approved

/cc @invincibleJai @florkbr

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
 
 export const ResourceStatus: React.FC<ResourceStatusProps> = ({
```
Comment (invincibleJai) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx:5]: we can have `ResourceStatus` as the default export over named one's and class names added in this component are like `co-resource-item__resource-status`  where `co` is for the console.   As we migrate to SDK shall we handle these as well?

cc @andrewballantyne

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export * from './utils/resource-status';
```
Comment (invincibleJai) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/index.ts:1]: if `resource-status` is default export can just export that and shall we expose it from SDK as well?

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx (hunk 3 above)
Comment (andrewballantyne) [reply frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx:5]: Agreed on the default export front.

The renaming of classNames is weird the way Console is coded... some of these classes belong to the global css file and we could dupe them, but I wouldn't "move" (rename) them. Once we are away from OpenShift Console and in the SDK repo, we'll need to get access to these classes so I think we have to dupe them and co-locate them -- which we might as well do as part of this PR. 

Good catch.

**Comment by fhlavac** [comment]
@invincibleJai @andrewballantyne 
updated the exports and also moved the css classes since they don't seem to be used anywhere else

File: frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts [hunk 5]
```diff
@@ -74,6 +75,8 @@ export const useListPageFilter: UseListPageFilter = require('@console/internal/c
   .useListPageFilter;
 export const ResourceLink: React.FC<ResourceLinkProps> = require('@console/internal/components/utils/resource-link')
   .ResourceLink;
+export const ResourceStatus: React.FC<ResourceStatusProps> = require('@console/dynamic-plugin-sdk/src/app/components/utils/resource-status')
+  .default;
```
Comment (andrewballantyne) [inline frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts:79]: Just directly export the component... no need to require internally to get the component.

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        isRead
+        data-test="resource-status"
+      >
+        {children}
+      </Badge>
+    </span>
+  );
+};
+
+export type ResourceStatusProps = {
```
Comment (andrewballantyne) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx:40]: No need to export the props of a component -- also could we have this at the top of the file too -- I think it's easier to read then putting it at the bottom?

```
// outside of this file, you can infer the props type from the component
type ResourceStatusProps = React.ComponentProps<typeof ResourceStatus>;

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { Badge } from '@patternfly/react-core';
+import * as classNames from 'classnames';
+import '../_resource.scss';
```
Comment (invincibleJai) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/resource-status.tsx:4]: not sure if we have a convention of the file name with `_*.scss`, and if these styles are only applicable to this component then can have style file name same as component one and at same level

**Comment by andrewballantyne** [comment]
Serverless Operator appears to be missing, the tests are being disabled in https://github.com/openshift/console/pull/10786
